---
title: "R Notebook"
output: html_notebook
editor_options: 
  chunk_output_type: console
---
# This is 16S data from O. Williamson SCTLD juvenile coral experiment. 

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
#16S First pass

```{r}  
knitr::opts_chunk$set(warning=FALSE, message=FALSE)


library(plyr) 
library(dplyr)
library(ggplot2)
library(ggpubr)
library(lmerTest)
library(car)
library(emmeans)
library(gridExtra)
library(multcomp)
library(reshape)
library(factoextra)
library(reshape2)
library(vegan) 
library(pairwiseAdonis)
library("scales")
packageVersion("scales")
library(RColorBrewer)
library(colorRamps)
library(devtools)
library(phyloseq)
library(readr)
library(vegan)
library(ape)
library(geosphere)
library(ade4)
library(microbiome)  
library(knitr)
library(parzer)

```

#Load in data and get into phyloseq (save RData file)
```{r}       

#parzer package to deal with coordinates
# ross coordinates how to process https://github.com/jrcunning/CBASS_FL_Acer/blob/main/data/genotype_metadata_raw.csv

#read in sample data and merge metadata
metadata<-read.csv("SCTLD_metadata_20240515.csv") 
  metadata$species<-as.factor(as.character(metadata$species))
  metadata$time.point<-as.factor(as.character(metadata$time.point))
  metadata$treatment<-as.factor(as.character(metadata$treatment))
  metadata$plug.id<-as.factor(as.character(metadata$plug.id))
  metadata$tank<-as.factor(as.character(metadata$tank))

#Make phyloseq Object
sam0 <- metadata
#sam1 <- as.matrix(sam0[, -1])
sam1 <- as.matrix(sam0)
rownames(sam1) <- sam0$sample_name
sam <- sample_data(data.frame(sam1))
#write.csv(sam, "sam.csv")

#load in OTU: abundance_table_100.shared.txt for counts table
OTU<-read.table("Results/main/details/abundance_table_100.shared.txt", sep='', header=T)

OTU <- OTU %>%
  mutate(Group = gsub("_", ".", Group)) %>%
  mutate(Group = gsub("S\\d{3}\\.L001", "", Group)) %>%
    mutate(Group = gsub("S\\d{2}\\.L001", "", Group)) %>%
  mutate(Group = gsub("\\.$", "", Group))

#otu
otu1 <- as.matrix(OTU[, -(1:3)]) # remove first col "label"
# replace the "Group" col with actual sample IDs.
# make a file with the "group" ids and the sample ids for metadata
indexes<-read.csv("data/index_new.csv")   #upload the sample ids
otu2<-merge(indexes,OTU, by="Group") #add sample names
otu2 <- as.matrix(otu2[, -c(1,3:4)]) # remove first col "Group", and label and num OTU
otu2.df<-as.data.frame(otu2) #make df copy to make samp names as row names in matrix

rownames(otu3k2) <- otu3k2.df$sample_name
otu3k2 <- as.matrix(otu3k2[, -(1)]) # remove first col samplename
otu<-otu3k2

#now it works
## something is preventing phyloseq from taking otu3k2 as the otu table. but if you save and then reupload it works. 
write.csv(otu3k2,"otu3k2check.csv")
testOtu<-read.csv("otu3k2check.csv")
testOtu2 <- as.matrix(testOtu[, -(1)]) 
rownames(testOtu2)<- otu3k2.df$sample_name
otu <- otu_table(testOtu2, taxa_are_rows = FALSE)
#write.csv(testOtu,"testOtu.csv")
#tax table annotations_100_taxonomy.csv (edited to be in proper format with proper col names in excel (remove ";"))
TAX<- read.csv("data/20230804_genoswap_16S-pipeline_outputs/Results/main/details/annotations_100.taxonomy.csv", colClasses = "character") 
tax1 <- as.matrix(TAX[, -1], dimnames = list(TAX$OTU, colnames(TAX[-1])))
rownames(tax1) <- TAX$OTU
tax <- tax_table(tax1)

# Read the data into phyloseq
Bac.seq = phyloseq(otu, tax,sam) #THIS WORKS
Bac.seq
Bac.seq.df <- sample_data(Bac.seq)

#load your tre file FastTree_100.nwk
treefile<- read.tree("data/20230804_genoswap_16S-pipeline_outputs/Results/postprocessing/unifrac/FastTree_100.nwk")
phy_tree(Bac.seq) <- treefile
Bac.seq

save(Bac.seq, file = "data/RData/Bac.seq_phyloseq2500new.RData")
```



